src/librustc_trans/mir/rvalue.rs

```diff
@@ -599,7 +608,8 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
     }
 }
 
-pub fn rvalue_creates_operand<'tcx>(rvalue: &mir::Rvalue<'tcx>) -> bool {
+pub fn rvalue_creates_operand<'bcx, 'tcx>(mir: &mir::Mir<'tcx>, bcx: Block<'bcx, 'tcx>,
```

Comment:
Pass a `BlockAndBuilder` here and use its `tcx` and `monomorphize` methods instead. `bcx.with_block` does work to make sure blocks are in suitable condition for old-trans and should be avoided at all costs unless you’re calling back into old-trans machinery.